src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -0,0 +1,426 @@
+                        injectData.append(dict(name=fileInfo['lfn'], scope=self.scope,
+                                               bytes=fileInfo['size'], state="A",
+                                               adler32=fileInfo['checksum']['adler32']))
+
+                    if self.rucio.createReplicas(rse=rseName, files=injectData, block=block):
+                        logging.info("Successfully inserted %d files on block %s", len(listLfns), block)
+                        self._updateLFNState(listLfns)
+        return
+
+    def _updateLFNState(self, listLfns, recovery=False):
+        """
+        Given a list of LFNs, update their state in dbsbuffer table.
+        :param listLfns: list of LFNs
+        :param recovery: True if we are recovering previously injected files
+        :return: nothing
+        """
+        if not listLfns:
+            return
+        try:
+            self.setStatus.execute(listLfns, 1)
+        except Exception as ex:
+            # save it to try to inject them again in the next cycle
+            self.filesToRecover.extend(listLfns)
+            if 'Deadlock found' in str(ex) or 'deadlock detected' in str(ex):
+                logging.error("Deadlock during file status update. Retrying again in the next cycle.")
+                self.filesToRecover.extend(listLfns)
+            else:
+                msg = "Failed to update file status in the database, reason: %s" % str(ex)
+                logging.error(msg)
+                raise RucioInjectorException(msg)
+        else:
+            if recovery:
+                self.filesToRecover = []
+
+    def closeBlocks(self):
+        """
+        Close any blocks that have been migrated to global DBS
+        """
+        logging.info("Starting closeBlocks method")
+
+        # in short, dbsbuffer_file.in_phedex = 1 AND dbsbuffer_block.status = 'InDBS'
+        migratedBlocks = self.getMigrated.execute()
+        ### FIXME the data format returned by this DAO
+        for location in migratedBlocks:
+            for container in migratedBlocks[location]:
+                for block in migratedBlocks[location][container]:
+                    if self.rucio.closeBlockContainer(block):
+                        self.setBlockClosed.execute(block)
+                    else:
+                        logging.error("Failed to close block: %s. Will retry again later.", block)
+
+    def deleteBlocks(self):
+        """
+        _deleteBlocks_
+        Find deletable blocks, then decide if to delete based on:
+        Is there an active subscription for dataset or block ?
+          If yes => set deleted=2
+          If no => next check
+        Has transfer to all destinations finished ?
+          If yes => request block deletion, approve request, set deleted=1
+          If no => do nothing (check again next cycle)
+        """
+        raise NotImplementedError
+
+    def deleteBlocksPhEDExCalls(self, location, blocksToDelete):
```

Comment:
I'm guessing this is just here to show logic of what we do now?

Reply:
Yes, I'm deleting it before pushing it in. Logic remains in PhEDExInjector, if needed.